vesuvius-c.h

```diff
@@ -820,4 +820,3366 @@ void reset_mesh_origin_to_roi(TriangleMesh *mesh, const RegionOfInterest *roi) {
     }
 }
 
+
+//vesuvius notes:
+// - when passing pointers to a _new function in order to fill out fields in the struct (e.g. mesh_new)
+//   the struct will take ownership of the pointer and the pointer shall be cleaned up in the _free function.
+//   The caller loses ownership of the pointer
+// - index order is in Z Y X order
+// - a 0 / SUCCESS return code indicates success for functions that do NOT return a pointer
+
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+#include <stdint.h>
+#include <math.h>
+#include <float.h>
+#include <stdbool.h>
+#include <assert.h>
+
+
+#include <sys/stat.h>
+#include <string.h>
+#include <errno.h>
+#include <curl/curl.h>
+
+
+#ifndef VESUVIUS_HEADER_ONLY
+  #ifndef VESUVIUS_STATIC_LIB
+    #ifndef VESUVIUS_DYNAMIC_LIB
+      #define VESUVIUS_HEADER_ONLY
+    #endif
+  #endif
+#endif
+
+
+#ifdef VESUVIUS_HEADER_ONLY
+  #define PUBLIC static inline __attribute__((visibility("hidden")))
+  #define PRIVATE static inline __attribute__((visibility("hidden")))
+#elifdef VESUVIUS_STATIC_LIB
+  #define PUBLIC static inline __attribute__((visibility("default")))
+  #define PRIVATE static inline __attribute__((visibility("hidden")))
+#elifdef VESUVIUS_DYNAMIC_LIB
+  #ifdef _WIN32
+    #define PUBLIC __declspec(dllexport)
+    #define PRIVATE static inline
+  #else
+    #define PUBLIC  __attribute__((visibility("default")))
+    #define PRIVATE static inline __attribute__((visibility("hidden")))
+  #endif
+#else
+  #define PUBLIC static inline
+  #define PRIVATE static inline
+#endif
+
+typedef enum errcode {
+  SUCCESS = 0,
```

Comment:
you might put a wart on these - VESUVIUS_SUCCESS, VESUVIUS_FAILURE - just in case somebody else tries to use the reserved words.